### ISSUE TITLE
GraphQl API creation failed

### ISSUE THREAD

**Issue by psycura** [aws-amplify/amplify-cli#99]
**Do you want to request a feature or report a bug?**
Bug
**What is the current behavior?**
After initialisation of GraphQl api, trying to do amplify push, but receive error:

CREATE_FAILED               ---------                AWS::CloudFormation::Stack Mon Sep 03 2018 16:13:56 GMT+0300 (Israel Daylight Time) Template format error: Parameter count 170 is greater than max allowed 60

my schema.graphql:

```
`type User  @model{
    id: ID!
    categories:[Category] @connection(name:"UserCategories")
    company: String!
    email: String!
    firstName: String!
    lastName: String!
    logo:String
    site:String!
    role:Int!
    type:String!
    package:Package! @connection(name:"UserPackage")
    customers:[Customer] @connection(name:"UserCustomers")
    resources:[Resource] @connection(name:"UserResources")
    proposals:[Proposal] @connection(name:"UserProposals")
    deliverables:[Deliverable] @connection(name:"UserDeliverables")
}

type Package @model{
    id:ID!
    deleted:Boolean!
    name:String!
    price:Float!
    type:Int!
    users: [User] @connection(name:"UserPackage")
}

type Category  @model{
    id:ID!
    name:String!
    user: User! @connection(name:"UserCategories")
}

type ContactPerson @model{
    company:String!
    email:String!
    firstName:String!
    id:ID!
    lastName:String!
    type:String!
    customer:Customer! @connection(name:"CustomerContactPersons")
    proposals:[Proposal] @connection(name:"ContactPersonProposals")
}

type Customer @model{
    id:ID!
    logo:String
    name:String!
    user:User! @connection(name:"UserCustomers")
    contactPersons:[ContactPerson] @connection(name:"CustomerContactPersons")
    proposals:[Proposal] @connection(name:"CustomerProposals")
}

type Resource @model{
    category:String!
    cost:Float!
    description:String!
    id:ID!
    name:String!
    price:Float!
    user:User! @connection(name:"UserResources")
    workItems:[WorkItem] @connection(name:"WorkItemResources")
}

type Proposal @model{
    createdAt: AWSDateTime!
    company:Customer! @connection(name:"CustomerProposals")
    contactPerson:ContactPerson! @connection(name:"ContactPersonProposals")
    id:ID!
    owner:User! @connection(name:"UserProposals")
    parameters:ProposalParameters! @connection(name:"ProposalParameters")
    sections:AWSJSON!
    deliverables:[Deliverable] @connection(name:"ProposalDeliverables")
}

type ProposalParameters @model{
    proposalId:ID!
    aroDate:AWSDateTime!
    startDate:AWSDateTime!
    hoursPerDay:Float!
    proposal: Proposal! @connection(name:"ProposalParameters")
}

type Deliverable @model{
    actualCost: Float!
    actualPrice:Float!
    cost:Float!
    description:String!
    estimatedCost:Float!
    estimatedWork:Float!
    id:ID!
    name:String!
    owner:User! @connection(name:"UserDeliverables")
    price:Float!
    proposal:Proposal! @connection(name:"ProposalDeliverables")
    deliveryItems:[DeliveryItem] @connection(name:"DeliverableDeliveryItems")
    workItems:[WorkItem] @connection(name:"DeliverableWorkItems")
}

type DeliveryItem @model{
    acceptance:String!
    deliveryTime:AWSDateTime!
    id:ID!
    name:String!
    deliverable:Deliverable! @connection(name:"DeliverableDeliveryItems")
}

type WorkItem @model{
    actualPrice:Float!
    actualStartTime:AWSDateTime
    startTime:AWSDateTime!
    actualWork:Float!
    category:String!
    description:String!
    title:String!
    type:String!
    work:String!
    duration:Float!
    parentId:ID!
    children:[WorkItem]
    estimatedCost:Float!
    estimatedStartTime:AWSDateTime
    estimatedEndTime:AWSDateTime
    estimatedPrice:Float!
    estimatedWork:Float!
    expanded:Boolean!
    id:ID!
    isVisible:Boolean!
    price:Float!
    resource:Resource! @connection(name:"WorkItemResources")
    deliverable:Deliverable! @connection(name:"DeliverableWorkItems")
}

`
```

**Comment by ChhorvornBradley** [comment]
I ran into this as well, I had the three items from the demo and two additional @models and one @searchable and it had 87 parameters, current workaround was making a smaller appsync but that really reduces the efficiency of this tool

**Comment by psycura** [comment]
So this is amplify limitations or appsync?
An what they call parameteres

**Comment by kaustavghosh06** [comment]
@psycura This is Cloudformation template limitations so, in turn, amplify limitations. We'll be looking into solving this problem.

**Comment by psycura** [comment]
So what i can do right now?
Create and connect appsync API manually, or you can recommend another approach?

**Comment by pigmanbear** [comment]
#11 
Have a look at the documentation. You can potentially workaround by extending the graphql-transformer and creating nested stacks. As mentioned in that thread, you will need a dependency on the schema creation before the creation of the resolvers, etc. 

I haven't had chance to work on it  enough to submit any changes, but the idea is: 

``` 
 "apiName": {
            "Type": "AWS::CloudFormation::Stack",
            "Properties": {
                "TemplateURL": "https://s3.amazonaws.com/{bucket}/amplify-cfn-templates/api/cloudformation-template.json",
                "Parameters": {
                    "AppSyncApiName": "Name",
                    "AuthCognitoUserPoolId": {
                        "Fn::GetAtt": [
                            "{cognitoPool}",
                            "Outputs.UserPoolId"
                        ]
                    },
                    "schemaGraphql": "s3://{bucket}/amplify-appsync-files/schema.graphql.1535083546157"
                },
                "Resources": {
                    "apiNoteModel": {
                        "Type": "AWS::CloudFormation::Stack",
                        "Properties": {
                            "TemplateURL": "https://s3.amazonaws.com/{bucket}/amplify-cfn-templates/api/models/note_model_cloudformation-template.json",
                            "Parameters": {
                                "AppSyncApiName": "rdntitle",
                                "AuthCognitoUserPoolId": {
                                    "Fn::GetAtt": [
                                        "{cognitoPool}",
                                        "Outputs.UserPoolId"
                                    ]
                                },
                                "QueryGetPrivateNoteRequest": "s3://{bucket}/amplify-appsync-files/Query.getPrivateNote.request.1535083546157",
                                "MutationUpdatePrivateNoteRequest": "s3://{bucket}/amplify-appsync-files/Mutation.updatePrivateNote.request.1535083546157",
                                "MutationUpdatePrivateNoteResponse": "s3://{bucket}/amplify-appsync-files/Mutation.updatePrivateNote.response.1535083546157",
                                "QueryListPrivateNotesRequest": "s3://{bucket}/amplify-appsync-files/Query.listPrivateNotes.request.1535083546157",
                                "QueryListPrivateNotesResponse": "s3://{bucket}/amplify-appsync-files/Query.listPrivateNotes.response.1535083546157",
                                "MutationCreatePrivateNoteRequest": "s3://{bucket}/amplify-appsync-files/Mutation.createPrivateNote.request.1535083546157"
                            }
                        }
                    },
                    "apiTaskModel": {
                        "Type": "AWS::CloudFormation::Stack",
                        "Properties": {
                            "TemplateURL": "https://s3.amazonaws.com/{bucket}/amplify-cfn-templates/api/models/task_model_cloudformation-template.json",
                            "Parameters": {
                                "AppSyncApiName": "rdntitle",
                                "AuthCognitoUserPoolId": {
                                    "Fn::GetAtt": [
                                        "{cognitoPool}",
                                        "Outputs.UserPoolId"
                                    ]
                                },
                                "MutationUpdateTaskResponse": "s3://{bucket}/amplify-appsync-files/Mutation.updateTask.response.1535083546157",
                                "MutationDeleteTaskRequest": "s3://{bucket}/amplify-appsync-files/Mutation.deleteTask.request.1535083546157",
                                "MutationCreatePrivateNoteResponse": "s3://{bucket}/amplify-appsync-files/Mutation.createPrivateNote.response.1535083546157",
                                "MutationCreateTaskRequest": "s3://{bucket}/amplify-appsync-files/Mutation.createTask.request.1535083546157",
                                "MutationDeletePrivateNoteResponse": "s3://{bucket}/amplify-appsync-files/Mutation.deletePrivateNote.response.1535083546157",
                                "MutationDeleteTaskResponse": "s3://{bucket}/amplify-appsync-files/Mutation.deleteTask.response.1535083546157",
                                "QueryGetPrivateNoteResponse": "s3://{bucket}/amplify-appsync-files/Query.getPrivateNote.response.1535083546157",
                                "MutationUpdateTaskRequest": "s3://{bucket}/amplify-appsync-files/Mutation.updateTask.request.1535083546157",
                                "MutationDeletePrivateNoteRequest": "s3://{bucket}/amplify-appsync-files/Mutation.deletePrivateNote.request.1535083546157",
                                "QueryGetTaskRequest": "s3://{bucket}/amplify-appsync-files/Query.getTask.request.1535083546157",
                                "MutationCreateTaskResponse": "s3://{bucket}/amplify-appsync-files/Mutation.createTask.response.1535083546157",
                                "QueryListTasksResponse": "s3://{bucket}/amplify-appsync-files/Query.listTasks.response.1535083546157",
                                "QueryListTasksRequest": "s3://{bucket}/amplify-appsync-files/Query.listTasks.request.1535083546157",
                                "QueryGetTaskResponse": "s3://{bucket}/amplify-appsync-files/Query.getTask.response.1535083546157"
                            }
                        }
                    }
                }
            },
```

And then in the CFTs for the Models you would '          
"DependsOn": "GraphQLSchema" for the creation of the resolvers. 

Making updates as part of the toolchain is essential of course and still working on getting it going and keeping a close eye on the repo!

**Comment by mikeparisstuff** [comment]
This is actively being worked on and should be in PR by the end of the week. In the future we are going to break the transform out into a nested stack instead of a single template to get around these cloudformation limits.

**Comment by mikeparisstuff** [comment]
While investigating the nested stack solution I found a quicker solution that should stop the immediate pain. I expect the next limit we hit will be the CloudFormation resource limit of 200, but this should work while we work on the nested stack solution.

**Comment by psycura** [comment]
Thank you, i saw that new version of cli is available - does this release include this fix?

**Comment by kaustavghosh06** [comment]
Fixed pushed with latest npm publish -> 0.1.18
